pkg/api/types.go

```diff
@@ -469,6 +469,7 @@ type MasterProfile struct {
 	ImageRef                 *ImageReference   `json:"imageReference,omitempty"`
 	CustomFiles              *[]CustomFile     `json:"customFiles,omitempty"`
 	AvailabilityProfile      string            `json:"availabilityProfile"`
+	PlatformFaultDomainCount *int              `json:"platformFaultDomainCount"`
```

Comment:
Can you explain why we don't want to carry this over to `vlabs` (and by extension, to persist it into the `apimodel.json`)?

Reply:
I guess I was considering it a hidden implementation detail. Would adding it in `vlabs` allow users to override it in a source apimodel? Because I think we don't want that.

Reply:
I was thinking if we put it in there during the first upgrade or scale (basically, once the cluster has been created the value can effectively be converted to a static int), then each successive upgrade or scale will have the value there conveniently without having to do the API calls.

Reply:
Consider the above comments as food for thought as this evolves.